apps/api/src/app/step-schemas/usecases/generate-preview/generate-preview-use-case.ts

```diff
@@ -0,0 +1,118 @@
+    const issues: Record<string, ControlPreviewIssue[]> = this.validateControlValues(command);
+    const hydratedPayload = addHydrationValuesToPayload(command);
+    const { workflowId, stepId, stepType } = await this.getWorkflowUserIdentifierFromWorkflowObject(command);
+    const executeOutput = await this.executePreviewUsecase(hydratedPayload, command, workflowId, stepId);
+
+    return buildResponse(issues, executeOutput, stepType);
+  }
+
+  private async executePreviewUsecase(
+    hydratedPayload: Record<string, unknown>,
+    command: GeneratePreviewCommand,
+    workflowId: string,
+    stepId: string
+  ) {
+    return await this.legacyPreviewStepUseCase.execute(
+      PreviewStepCommand.create({
+        payload: hydratedPayload,
+        controls: command.generatePreviewRequestDto.controlValues || {},
+        environmentId: command.user.environmentId,
+        organizationId: command.user.organizationId,
+        stepId,
+        userId: command.user._id,
+        workflowId,
+      })
+    );
+  }
+
+  private async getWorkflowUserIdentifierFromWorkflowObject(command: GeneratePreviewCommand) {
+    const workflowResponseDto = await this.getWorkflowUseCase.execute({
+      identifierOrInternalId: command.workflowId,
+      user: command.user,
+    });
+    const { workflowId, steps } = workflowResponseDto;
+    const step = steps.find((stepDto) => stepDto.stepUuid === command.stepUuid);
+    if (!step) {
+      throw new BadRequestException(`Step id found for ${command.stepUuid}`);
+    }
+
+    return { workflowId, stepId: step.slug, stepType: step.type };
+  }
+
+  private validateControlValues(command: GeneratePreviewCommand) {
+    return this.validator.searchAndValidatePlaceholderExistence(
+      command.generatePreviewRequestDto.validationStrategies,
+      command.generatePreviewRequestDto.controlSchema,
+      command.generatePreviewRequestDto.payloadValues,
+      command.generatePreviewRequestDto.controlValues
+    );
+  }
+}
+
+function addHydrationValuesToPayload(command: GeneratePreviewCommand): Record<string, unknown> {
+  const dto = command.generatePreviewRequestDto;
+  let payloadFromDto = dto.payloadValues || {};
+  for (const controlValueKey in dto.controlValues) {
+    if (dto.controlValues.hasOwnProperty(controlValueKey)) {
+      const hydratedValue = new CreateMockPayloadUseCase().execute({ dto, controlValueKey });
+      console.log('hydratedValue', hydratedValue);
+      payloadFromDto = mergeJsonObjects(payloadFromDto, hydratedValue);
+    }
+  }
+
+  return payloadFromDto;
+}
+function buildResponse(
+  issues: Record<string, ControlPreviewIssue[]>,
+  executionOutput: ExecuteOutput,
+  stepType: StepTypeEnum
+): GeneratePreviewResponseDto {
+  return {
+    issues,
+    result: {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      preview: executionOutput.outputs as any,
+      type: stepType as unknown as ChannelTypeEnum,
+    },
+  };
+}
+function mergeJsonObjects(
```

Comment:
How about using [this](https://github.com/novuhq/novu/blob/a305ec690cbdd7b89e636379cbfd36657d55a662/packages/providers/src/utils/deepmerge.utils.ts#L1) or [lodash.merge](https://lodash.com/docs/4.17.15#merge). Lodash is already used across the codebase.